src/onnx/onnx.cpp

```diff
@@ -751,6 +754,66 @@ struct onnx_parser
         return add_bias(args, l1, 1);
     }
 
+    void
+    tune_padding_to_symmetric(int64_t& left, int64_t& right, const int stride, int64_t& s_start)
+    {
+        s_start = 0;
+        if(left > right)
+        {
+            right   = left;
+            s_start = 0;
+        }
+        else if(left < right)
+        {
+            auto diff = right - left;
+            s_start   = (diff + stride - 1) / stride;
+            left      = left + s_start * stride;
+            right     = left;
+        }
+
+        return;
+    }
+
+    template <class Op>
+    void tune_padding_size(Op& op,
+                           std::vector<int64_t>& padding,
+                           int count_include_pad,
+                           std::vector<int64_t>& s_start)
+    {
+        // maxpooling or count_include_pad is 1, no change is required.
+        if(op.mode == "max" or count_include_pad == 1)
+        {
+            return;
+        }
+
+        // check assymetric padding
+        bool asym_padding = false;
+        assert(padding.size() % 2 == 0);
```

Comment:
I made a function in PR #554 that you can use to check for asym_padding

Reply:
fixed